### DISCUSSION TITLE
--hls-segment-ignore-names

### DISCUSSION THREAD

**Discussion by fenopa** [streamlink/streamlink#4944] [Q&A] [answered]
Hi, I added `--hls-segment-ignore-names "/ad/"` parameter to filter ad segments which have urls like this: `https://tv8-live.daioncdn.net/tv8/ad/x/x/x.ts` but it doesn't seem to work. Does that parameter only look if the URL ends with that value? Why just that? Why not a "contains" check?
Anyway is there way to achieve this and ignore these ad segments?
Thanks.

**Comment by bastimeyer** [comment, marked as answer]
This is unfortunately not possible. When the parameter was added, the HLS implementation didn't even have a proper filtering system. That came later. And while the segment names are tested in a built regex, the values you pass are all regex-escaped.
https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/stream/hls.py#L80-L85

A proper implementation without introducing any breaking API changes would be adding a new parameter `--hls-segment-ignore-name` which can be set multiple times instead of having a comma-separated list of names, and which receives regexes as values that get matched against the HLS segment names.

**Reply by fenopa** [reply]
Thank you for the answer. You say there is a "proper filtering system" now, but only way to apply that is to modify the relevant plugin, right? Can you give some examples how this simple filtering can be applied and achieved with it? Thanks again.

**Reply by bastimeyer** [reply]
Plugins would need to use a custom HLSStreamWriter subclass with a modified `should_filter_sequence()` method.
https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/stream/hls.py#L201-L202

Take a look at other plugins which do that, like Twitch, Afreeca, AbemaTV, Pluto, etc:
- https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/plugins/twitch.py#L156-L157
- https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/plugins/afreeca.py#L19
- https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/plugins/abematv.py#L33
- https://github.com/streamlink/streamlink/blob/5.0.1/src/streamlink/plugins/pluto.py#L24
